### ISSUE TITLE
documentation of SC2115 is incorrect

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#2717]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"):  online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
VAR="${PWD}"
rm -rf ${VAR}/*
```

#### Here's what shellcheck currently says:

SC2115: Use "${var:?}" to ensure this never expands to /* 
Exceptions: None.

#### Here's what I wanted or expected to see:

It looks like it is one exception (setting VAR to a known non-empty value), and also if we check VAR before usage [[ -z VAR ]] then this SC2115 will be pointless, another exception.
The third one could be the case when VAR is intentionally empty and we want to remove everything removable in / (yes, it is definitely unlikely or strange, but not impossible).
I know it is not that important, just a small correction in the documentation.
